### PR TITLE
Resolves #204 error message list

### LINKED ISSUE #204: Add section to user manual about log messages
Add a section to the user manual that explains the log messages, what the levels (e.g., ERROR, WARNING) mean and how to address them.

### PR DESCRIPTION
**Pull request by rockyallen**
Please don't accept this! I just want to check the layout with you before spending time populating it.

### CODE REVIEW

**Comment by mojavelinux** [comment]
I do like it, though I wonder if the category of the message should be a separate column. I don't think the "asciidoctor" prefix is necessary on the message since that is always the same.

**Comment by rockyallen** [comment]
Table updated - comments added in italics where I don't understand.

File: docs/_includes/errormessages.adoc [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+|<<using-the-command-line-interface>>
+
+|ERROR
+|include file has illegal reference to ancestor of jail, auto-recovering
+|
+|<<running-asciidoctor-securely>>
+
+|ERROR
+|IOError, %(input file and output file cannot be the same: #outfile)
+|__I cant produce this message. asciidoctor -o file.adoc file.adoc overwrites file.adoc silently.__
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: We're missing a check for that case. I'll need to fix that. The case currently checked is:

```
asciidoctor -a outfilesuffix=.adoc file.adoc
```

File: docs/_includes/errormessages.adoc [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+|
+
+|ERROR
+|IOError, %(target directory does not exist: #{to_dir})
+|__I cant produce this message. asciidoctor -D doesnotexist file.adoc creates doesnotexist silently.__
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: This only happens when using Asciidoctor via the API, the mkdirs option is not set, and the target directory does not already exist.

File: docs/_includes/errormessages.adoc [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+====
+
+.Common Error Messages
+[options="header,breakable",cols="<20,<60,<50,<30"]
+|====
+|Category (1)|Message |Cause |See also
+
+|ERROR
+|input file <file> missing or cannot be read
+|Misspelled filename?
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: Let's write this as "check that the file exists and that the filename is not misspelled"

File: docs/_includes/errormessages.adoc [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+|<<include-uri>>
+
+|WARNING
+|could not retrieve contents of <x> at URI: <uri>
+|Web address not found.
+|
+
+|WARNING
+|could not retrieve image data from URI: <uri>
+|__cant reproduce, even with :data-uri: and non existent file__
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: Given the following AsciiDoc:

```
image::http://example.org/invalid-image.jpg[]
```

You can reproduce by running:

```
asciidoctor -a allow-uri-read -a data-uri sample.adoc
```

**Comment by mojavelinux** [comment]
Looking good! In some cases, the error message still has some Ruby code in it. Can you strip those down to just the message? (look for trailing if statements or IOError() wrappers).

**Comment by rockyallen** [comment]
Comments addressed, extra failure added, discovered  when I typed -b hhtml

File: docs/_includes/errormessages.adoc [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+Errors do not stop conversion, but the output document will almost certainly be wrong.
+
+Warnings do not stop conversion, but they indicate possible problems, and the output may not be what you were expecting.
+
+// Rare/unexplained
+
+////
+|WARNING
+|gem \'thread_safe\' is not installed. This gem is recommended when registering custom converters.
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: This warning occurs if you have registered a custom converter implementation and you try to use asciidoctor without first installing the thread_safe gem.

File: docs/_includes/errormessages.adoc [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+|
+|
+
+|WARNING
+|gem \'thread_safe\' is not installed. This gem is recommended when using custom backend templates.
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: This warning occurs if you use custom templates (`-T <template_dir>`) and you try to use asciidoctor without first installing the thread_safe gem.

File: docs/_includes/errormessages.adoc [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+|
+|
+
+|WARNING
+|file does not exist or cannot be read: #{path}) if opts[:warn_on_failure]
+|__not tested. I dont know when it could occur, apart from include which is already covered__
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: This happens if you specify `-a stylesheet=<file>` and `<file>` does not exist or is not readable (no permission).

File: docs/_includes/errormessages.adoc [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+// use <x,y,z> for anything else
+
+////
+Testing was carried out with
+Asciidoctor 1.5.5 [http://asciidoctor.org]
+Runtime Environment (ruby 2.3.1p112 (2016-04-26) [i386-linux-gnu]) (lc:UTF-8 fs:UTF-8 in:- ex:UTF-8)
+////
+
+The most common error and warning messages when using the asciidoctor command line (including asciidoctor-ant) are listed alphabetically in the table below.
+Messages from asciidoctorj or the asciidoctor API are not included.
```
Comment (mojavelinux) [inline docs/_includes/errormessages.adoc]: Add a comment to this line about asciidoctorj and asciidoctor API. I think that just confuses the reader for now, but it is important information for the documenter.

**Comment by mojavelinux** [comment]
A few more comments inline. I think we're almost to a first version.

**Comment by rockyallen** [comment]
Updated as comments

**Comment by mojavelinux** [comment]
I've made some revisions and rebased the PR into a single commit.

**Comment by mojavelinux** [comment]
Let's go for it! Thanks!

**Pull request merged**